src/framework/entity.js

```diff
@@ -150,6 +150,57 @@ Object.assign(pc, function () {
         system.removeComponent(this);
     };
 
+    /**
+     * @function
+     * @name pc.Entity#getComponent
+     * @description Get a component of specified type from the entity.
+     * @param {String} type The name of the component type to retrieve.
+     * @returns {pc.Component} A component of specified type if the entity has one, undefined if it doesn't.
+     * @example
+     * var entity = new pc.Entity();
+     * entity.addComponent("light"); // add new light component
+     * //...
+     * var light = entity.getComponent("light"); // get light component
+     */
+    Entity.prototype.getComponent = function (type) {
```

Comment:
Do we need this function? I mean, why do entity.getComponent("light") instead of just entity.light?

Reply:
I get that `entity.light` is easier. But when you don't know the `type`, I find it's useful to provide a clear and safe way of accessing components. So instead of writing `entity[type]`, which could return any random property or function on the entity, it's nice to be able to write `entity.getComponent(type)`. I believe other component-based engines like Unity provide both as well.

Reply:
Hmm, OK. I'm going to defer to @daredevildave on this since he originally designed the entity-component system. He should be able to take a look early next week. I hope that's OK. 😄

Reply:
Of course. Happy to let @daredevildave take a look. I understand it looks redundant to have two ways to access the same component and I probably could have given a better example in the documentation :)

Reply:
An alternative is that you refactor this PR to not have the Entity#getComponent function. I'd take the Entity#findComponent(s) functions. They do look very useful. And you can propose Entity#getComponent separately and @daredevildave can discuss with you when he's available. Up to you. 😃

Reply:
Sure. Will create separate PR for `Entity.getComponent`

Reply:
I've removed `Entity.getComponent` from this PR